src/lib.rs

```diff
@@ -4,85 +4,108 @@ use pgrx::prelude::*;
 use pgrx::shmem::*;
 use pgrx::spi::SpiResult;
 use pgrx::{debug1, error, pg_shmem_init, GucContext, GucFlags, GucRegistry, GucSetting};
-use serde::{Deserialize, Serialize};
-use std::collections::HashMap;
 use std::ffi::CStr;
+use std::time::Duration;
 
-// Max allocation params
+// Capacity params
+// IMPORTANT: All capacity values MUST be a power of 2. E.g. 2^8 = 256, 2^9 = 512, 2^10 = 1024
 
-const MAX_ENTRIES: usize = 8 * 1024;
-const MAX_CURRENCIES: usize = 1024;
+const MAX_ENTRIES: usize = 512;
+const MAX_CURRENCIES: usize = 64;
+const MAX_ID_PAIRS: usize = 1024;
+const CURRENCY_XUID_MAX_LEN: usize = 64;
 
 // Default Queries
 
 const DEFAULT_Q1_VALIDATION_QUERY: &CStr = cr#"SELECT count(table_name) = 2
 FROM information_schema.tables
 WHERE table_schema = 'plan' AND (table_name = 'currency' or table_name = 'fx_rate');"#;
 
-const DEFAULT_Q2_GET_CURRENCIES_IDS_QUERY: &CStr =
-    cr#"SELECT min(c.id), max(c.id) FROM plan.currency c"#;
-
-const DEFAULT_Q3_GET_CURRENCIES_ENTRY_COUNT: &CStr = cr#"SELECT cu.id,
-LOWER(cu.xuid),
-(SELECT count(*) rate_count FROM plan.fx_rate fr WHERE fr.currency_id = cu.id)
+const DEFAULT_Q2_GET_CURRENCIES_XUID_INIT: &CStr = cr#"SELECT cu.id, LOWER(cu.xuid)
```

Comment:
lets remove LOWER(cu.xuid) and in the functions where we are getting rate by XUID